pkg/projects/resource_project_user.go

```diff
@@ -0,0 +1,214 @@
+		projectUsers := v.(*schema.Set).List()
+		if len(projectUsers) == 0 {
+			return projectKey, users, nil
+		}
+
+		for _, projectUser := range projectUsers {
+			id := projectUser.(map[string]interface{})
+
+			user := User{
+				Name:  id["name"].(string),
+				Roles: castToStringArr(id["roles"].(*schema.Set).List()),
+			}
+			users.Members = append(users.Members, user)
+		}
+	}
+
+	return projectKey, users, nil
+}
+
+var packUsers = func(d *schema.ResourceData, key string, users *[]User) []error {
+	log.Printf("[DEBUG] packUsers")
+
+	setValue := mkLens(d)
+
+	var projectUsers []interface{}
+
+	for _, user := range *users {
+		log.Printf("[TRACE] %+v\n", user)
+		projectUser := map[string]interface{}{
+			"name":  user.Name,
+			"roles": user.Roles,
+		}
+
+		projectUsers = append(projectUsers, projectUser)
+	}
+
+	log.Printf("[TRACE] %s\n", key)
+	log.Printf("[TRACE] %+v\n", projectUsers)
+
+	errors := setValue(key, projectUsers)
+
+	return errors
+}
+
+var getProjectsUsersUrl = func(projectKey string, id string) string {
+	return fmt.Sprintf(projectUsersUrl, projectKey) + id
+}
+
+var readUsers = func(projectKey string, m interface{}) ([]User, error) {
+	log.Println("[DEBUG] readUsers")
+
+	users := Users{}
+
+	_, err := m.(*resty.Client).R().SetResult(&users).Get(fmt.Sprintf(projectUsersUrl, projectKey))
+	if err != nil {
+		return nil, err
+	}
+
+	log.Printf("[TRACE] %+v\n", users)
+
+	return users.Members, nil
+}
+
+var updateUsers = func(projectKey string, terraformUsers Users, m interface{}) ([]User, error) {
+	log.Println("[DEBUG] updateUsers")
+	log.Printf("[TRACE] terraformUsers: %+v\n", terraformUsers)
+
+	projectUsers, err := readUsers(projectKey, m)
+	log.Printf("[TRACE] projectUsers: %+v\n", projectUsers)
+
+	usersToBeAdded := usersDifference(terraformUsers.Members, projectUsers)
+	log.Printf("[TRACE] usersToBeAdded: %+v\n", usersToBeAdded)
+	usersToBeUpdated := usersIntersection(terraformUsers.Members, projectUsers)
+	log.Printf("[TRACE] usersToBeUpdated: %+v\n", usersToBeUpdated)
+	usersToBeDeleted := usersDifference(projectUsers, terraformUsers.Members)
+	log.Printf("[TRACE] usersToBeDeleted: %+v\n", usersToBeDeleted)
+
+	var errs []error
+
+	for _, user := range append(usersToBeAdded, usersToBeUpdated...) {
+		log.Printf("[TRACE] %+v\n", user)
+		err := updateUser(projectKey, &user, m)
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	err = deleteUsers(projectKey, usersToBeDeleted, m)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	if errs != nil && len(errs) > 0 {
+		return nil, fmt.Errorf("failed to update users for project: %s", fmt.Sprintf("%s", errs))
+	}
+
+	return readUsers(projectKey, m)
+}
+
+var updateUser = func(projectKey string, user *User, m interface{}) error {
+	log.Println("[DEBUG] updateUser")
+
+	_, err := m.(*resty.Client).R().SetBody(user).Put(getProjectsUsersUrl(projectKey, user.Name))
+
+	return err
+}
+
+var deleteUsers = func(projectKey string, users []User, m interface{}) error {
```

Comment:
I would prefer an atomic `deleteUser` and then have this iterate over it. 

something to consider: What do you do if you can't delete all users from the project?